src/domain/registrations/searchPanel/SearchPanel.tsx

```diff
@@ -0,0 +1,129 @@
+import { css } from '@emotion/css';
+import classNames from 'classnames';
+import { IconHeart, IconSearch } from 'hds-react';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory, useLocation } from 'react-router';
+
+import Button from '../../../common/components/button/Button';
+import MultiSelectDropdown from '../../../common/components/multiSelectDropdown/MultiSelectDropdown';
+import SearchInput from '../../../common/components/searchInput/SearchInput';
+import { ROUTES } from '../../../constants';
+import useLocale from '../../../hooks/useLocale';
+import { OptionType } from '../../../types';
+import { useTheme } from '../../app/theme/Theme';
+import { EVENT_TYPE } from '../../event/constants';
+import useEventTypeOptions from '../../event/hooks/useEventTypeOptions';
+import {
+  getRegistrationSearchInitialValues,
+  getRegistrationSearchQuery,
+} from '../utils';
+import styles from './searchPanel.module.scss';
+
+type SearchState = {
+  eventType: EVENT_TYPE[];
+  text: string;
+};
+
+const SearchPanel: React.FC = () => {
+  const { t } = useTranslation();
+  const { theme } = useTheme();
+  const history = useHistory();
+  const location = useLocation();
+  const locale = useLocale();
+
+  const eventTypeOptions = useEventTypeOptions();
+
+  const [searchState, setSearchState] = React.useReducer(
+    (prevState: SearchState, updatedProperty: Partial<SearchState>) => ({
+      ...prevState,
+      ...updatedProperty,
+    }),
+    { eventType: [], text: '' }
+  );
```

Comment:
Noticed that this usage of useReducer is found in 3 places. Could be extracted to reusable hook. Something like `useReducerState`. Of course you can come up with better naming :)

```
 const [searchState, setSearchState] = useReducerState<SearchState>({
    eventType: [],
    text: '',
  });
```

```
const useReducerState = <T>(
  initialState: T
): [T & Partial<T>, React.Dispatch<Partial<T>>] => {
  return React.useReducer(
    (prevState: T, updatedProperty: Partial<T>) => ({
      ...prevState,
      ...updatedProperty,
    }),
    initialState
  );
};
```